### PR TITLE
Set a maximum number of accepted deadline miss the the AdvanceableRunner

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
As per the title, this PR adds the possibility to stop the advanceable runner if the number of deadline misses exceeds the accepted maximum

### CODE REVIEW

**Review by S-Dafarra — APPROVED** [review]

**Comment by GiulioRomualdi** [comment]
A test is failing

https://github.com/ami-iit/bipedal-locomotion-framework/actions/runs/6273110343/job/17035936038?pr=726#step:24:171


Maybe due to #724 ? 🤔  @traversaro

File: src/System/include/BipedalLocomotion/System/AdvanceableRunner.h [hunk 1]
```diff
@@ -314,16 +325,30 @@ AdvanceableRunner<_Advanceable>::run(std::optional<std::reference_wrapper<Barrie
...
+                    log()->error("{} - {} Experienced {} deadline misses. The maximum accepted "
+                                 "number is {}.",
+                                 logPrefix,
+                                 m_info.name,
+                                 m_info.deadlineMiss,
+                                 m_maximumNumberOfAcceptedDeadlineMiss);
+
+                    // we have to close the runner
+                    this->m_isRunning = false;
+                    break;
```
Comment (traversaro) [inline src/System/include/BipedalLocomotion/System/AdvanceableRunner.h:345]: This is done only when telemetry is enabled, is this intentional? Or perhaps know that deadlineMiss is used for something else we should change the logic?

**Comment by traversaro** [comment]
> A test is failing
> 
> https://github.com/ami-iit/bipedal-locomotion-framework/actions/runs/6273110343/job/17035936038?pr=726#step:24:171
> 
> Maybe due to #724 ? 🤔 @traversaro

Yes, it seems it was failing also there https://github.com/ami-iit/bipedal-locomotion-framework/pull/724 .

**Comment by traversaro** [comment]
> > A test is failing
> > https://github.com/ami-iit/bipedal-locomotion-framework/actions/runs/6273110343/job/17035936038?pr=726#step:24:171
> > Maybe due to #724 ? 🤔 @traversaro
> 
> Yes, it seems it was failing also there #724 .

Fixed by https://github.com/ami-iit/bipedal-locomotion-framework/pull/727 .

File: src/System/include/BipedalLocomotion/System/AdvanceableRunner.h (hunk 1 above)
Comment (GiulioRomualdi) [reply src/System/include/BipedalLocomotion/System/AdvanceableRunner.h:345]: > This is done only when telemetry is enabled, is this intentional

Before I used it just to print some info at the end of the application but probably now it's better to keep this always active. What do you think?

**Pull request merged**